app/controllers/application_controller.rb

```diff
@@ -0,0 +1,23 @@
+class ApplicationController < ActionController::Base
+
+  # before_action :require_login
+
+  def current_user
+    @current_user = User.find_by(id: session[:user_id])
+
+    # unless @current_user
+    #   # TODO: must be logged in to upvote
+    #   flash[:error] = "You must be logged in to see this page"
+    #   redirect_to root_path
+    #   return
+    # end
```

Comment:
You have a lot of commented out code in this project.  

It makes it a little confusing to read.  In the future you should clean this up if you have time before you turn things in.